// @radixdlt/alphanet-gateway-api-v0-sdk/dist/alphanet-gateway-api-v0-sdk.mjs
var nt = Object.defineProperty;
var rt = (t, e, n) => e in t ? nt(t, e, { enumerable: !0, configurable: !0, writable: !0, value: n }) : t[e] = n;
var y = (t, e, n) => (rt(t, typeof e != "symbol" ? e + "" : e, n), n);
const it = "https://alphanet.radixdlt.com/v0".replace(/\/+$/, "");
class ot {
  constructor(e = {}) {
    this.configuration = e;
  }
  set config(e) {
    this.configuration = e;
  }
  get basePath() {
    return this.configuration.basePath != null ? this.configuration.basePath : it;
  }
  get fetchApi() {
    return this.configuration.fetchApi;
  }
  get middleware() {
    return this.configuration.middleware || [];
  }
  get queryParamsStringify() {
    return this.configuration.queryParamsStringify || F;
  }
  get username() {
    return this.configuration.username;
  }
  get password() {
    return this.configuration.password;
  }
  get apiKey() {
    const e = this.configuration.apiKey;
    if (e)
      return typeof e == "function" ? e : () => e;
  }
  get accessToken() {
    const e = this.configuration.accessToken;
    if (e)
      return typeof e == "function" ? e : async () => e;
  }
  get headers() {
    return this.configuration.headers;
  }
  get credentials() {
    return this.configuration.credentials;
  }
}
const st = new ot();
class w {
  constructor(e = st) {
    y(this, "middleware");
    y(this, "fetchApi", async (e, n) => {
      let r = { url: e, init: n };
      for (const o of this.middleware)
        o.pre && (r = await o.pre({
          fetch: this.fetchApi,
          ...r
        }) || r);
      let i;
      try {
        i = await (this.configuration.fetchApi || fetch)(r.url, r.init);
      } catch (o) {
        for (const s of this.middleware)
          s.onError && (i = await s.onError({
            fetch: this.fetchApi,
            url: r.url,
            init: r.init,
            error: o,
            response: i ? i.clone() : void 0
          }) || i);
        if (i !== void 0)
          throw new dt(o, "The request failed and the interceptors did not return an alternative response");
      }
      for (const o of this.middleware)
        o.post && (i = await o.post({
          fetch: this.fetchApi,
          url: r.url,
          init: r.init,
          response: i.clone()
        }) || i);
      return i;
    });
    this.configuration = e, this.middleware = e.middleware;
  }
  withMiddleware(...e) {
    const n = this.clone();
    return n.middleware = n.middleware.concat(...e), n;
  }
  withPreMiddleware(...e) {
    const n = e.map((r) => ({ pre: r }));
    return this.withMiddleware(...n);
  }
  withPostMiddleware(...e) {
    const n = e.map((r) => ({ post: r }));
    return this.withMiddleware(...n);
  }
  async request(e, n) {
    const { url: r, init: i } = await this.createFetchParams(e, n), o = await this.fetchApi(r, i);
    if (o.status >= 200 && o.status < 300)
      return o;
    throw new ct(o, "Response returned an error code");
  }
  async createFetchParams(e, n) {
    let r = this.configuration.basePath + e.path;
    e.query !== void 0 && Object.keys(e.query).length !== 0 && (r += "?" + this.configuration.queryParamsStringify(e.query));
    const i = Object.assign({}, this.configuration.headers, e.headers);
    Object.keys(i).forEach((J) => i[J] === void 0 ? delete i[J] : {});
    const o = typeof n == "function" ? n : async () => n, s = {
      method: e.method,
      headers: i,
      body: e.body,
      credentials: this.configuration.credentials
    }, l = {
      ...s,
      ...await o({
        init: s,
        context: e
      })
    }, et = {
      ...l,
      body: at(l.body) || l.body instanceof URLSearchParams || ut(l.body) ? l.body : JSON.stringify(l.body)
    };
    return { url: r, init: et };
  }
  clone() {
    const e = this.constructor, n = new e(this.configuration);
    return n.middleware = this.middleware.slice(), n;
  }
}
function ut(t) {
  return typeof Blob < "u" && t instanceof Blob;
}
function at(t) {
  return typeof FormData < "u" && t instanceof FormData;
}
class ct extends Error {
  constructor(n, r) {
    super(r);
    y(this, "name", "ResponseError");
    this.response = n;
  }
}
class dt extends Error {
  constructor(n, r) {
    super(r);
    y(this, "name", "FetchError");
    this.cause = n;
  }
}
class _ extends Error {
  constructor(n, r) {
    super(r);
    y(this, "name", "RequiredError");
    this.field = n;
  }
}
const An = {
  csv: ",",
  ssv: " ",
  tsv: "	",
  pipes: "|"
};
function a(t, e) {
  const n = t[e];
  return n != null;
}
function F(t, e = "") {
  return Object.keys(t).map((n) => R(n, t[n], e)).filter((n) => n.length > 0).join("&");
}
function R(t, e, n = "") {
  const r = n + (n.length ? `[${t}]` : t);
  if (e instanceof Array) {
    const i = e.map((o) => encodeURIComponent(String(o))).join(`&${encodeURIComponent(r)}=`);
    return `${encodeURIComponent(r)}=${i}`;
  }
  if (e instanceof Set) {
    const i = Array.from(e);
    return R(t, i, n);
  }
  return e instanceof Date ? `${encodeURIComponent(r)}=${encodeURIComponent(e.toISOString())}` : e instanceof Object ? F(e, r) : `${encodeURIComponent(r)}=${encodeURIComponent(String(e))}`;
}
function Kn(t, e) {
  return Object.keys(t).reduce(
    (n, r) => ({ ...n, [r]: e(t[r]) }),
    {}
  );
}
function Mn(t) {
  for (const e of t)
    if (e.contentType === "multipart/form-data")
      return !0;
  return !1;
}
class d {
  constructor(e, n = (r) => r) {
    this.raw = e, this.transformer = n;
  }
  async value() {
    return this.transformer(await this.raw.json());
  }
}
class $n {
  constructor(e) {
    this.raw = e;
  }
  async value() {
  }
}
class Un {
  constructor(e) {
    this.raw = e;
  }
  async value() {
    return await this.raw.blob();
  }
}
class Wn {
  constructor(e) {
    this.raw = e;
  }
  async value() {
    return await this.raw.text();
  }
}
function zn(t) {
  let e = !0;
  return e = e && "state_version" in t, e;
}
function Bn(t) {
  return _t(t);
}
function _t(t, e) {
  return t == null ? t : {
    state_version: t.state_version
  };
}
function Hn(t) {
  if (t !== void 0)
    return t === null ? null : {
      state_version: t.state_version
    };
}
const Gn = {
  EcdsaSecp256k1: "EcdsaSecp256k1",
  EddsaEd25519: "EddsaEd25519"
};
function h(t) {
  return ft(t);
}
function ft(t, e) {
  return t;
}
function Ln(t) {
  return t;
}
function Qn(t) {
  let e = !0;
  return e = e && "key_type" in t, e = e && "signature_hex" in t, e;
}
function pt(t) {
  return k(t);
}
function k(t, e) {
  return t == null ? t : {
    key_type: h(t.key_type),
    signature_hex: t.signature_hex
  };
}
function x(t) {
  if (t !== void 0)
    return t === null ? null : {
      key_type: t.key_type,
      signature_hex: t.signature_hex
    };
}
function Xn(t) {
  let e = !0;
  return e = e && "key_type" in t, e = e && "signature_hex" in t, e;
}
function lt(t) {
  return P(t);
}
function P(t, e) {
  return t == null ? t : {
    key_type: h(t.key_type),
    signature_hex: t.signature_hex
  };
}
function E(t) {
  if (t !== void 0)
    return t === null ? null : {
      key_type: t.key_type,
      signature_hex: t.signature_hex
    };
}
function yt(t) {
  return mt(t);
}
function mt(t, e) {
  if (t == null)
    return t;
  switch (t.key_type) {
    case "EcdsaSecp256k1":
      return { ...k(t), key_type: "EcdsaSecp256k1" };
    case "EddsaEd25519":
      return { ...P(t), key_type: "EddsaEd25519" };
    default:
      throw new Error(`No variant of Signature exists with 'key_type=${t.key_type}'`);
  }
}
function St(t) {
  if (t !== void 0) {
    if (t === null)
      return null;
    switch (t.key_type) {
      case "EcdsaSecp256k1":
        return x(t);
      case "EddsaEd25519":
        return E(t);
      default:
        throw new Error(`No variant of Signature exists with 'key_type=${t.key_type}'`);
    }
  }
}
function Yn(t) {
  let e = !0;
  return e = e && "key_type" in t, e = e && "recoverable_signature" in t, e;
}
function Zn(t) {
  return I(t);
}
function I(t, e) {
  return t == null ? t : {
    key_type: h(t.key_type),
    recoverable_signature: pt(t.recoverable_signature)
  };
}
function ht(t) {
  if (t !== void 0)
    return t === null ? null : {
      key_type: t.key_type,
      recoverable_signature: x(t.recoverable_signature)
    };
}
function vn(t) {
  let e = !0;
  return e = e && "key_type" in t, e = e && "key_hex" in t, e;
}
function bt(t) {
  return V(t);
}
function V(t, e) {
  return t == null ? t : {
    key_type: h(t.key_type),
    key_hex: t.key_hex
  };
}
function C(t) {
  if (t !== void 0)
    return t === null ? null : {
      key_type: t.key_type,
      key_hex: t.key_hex
    };
}
function jn(t) {
  let e = !0;
  return e = e && "key_type" in t, e = e && "public_key" in t, e = e && "signature" in t, e;
}
function tr(t) {
  return D(t);
}
function D(t, e) {
  return t == null ? t : {
    key_type: h(t.key_type),
    public_key: bt(t.public_key),
    signature: lt(t.signature)
  };
}
function gt(t) {
  if (t !== void 0)
    return t === null ? null : {
      key_type: t.key_type,
      public_key: C(t.public_key),
      signature: E(t.signature)
    };
}
function Ot(t) {
  return Nt(t);
}
function Nt(t, e) {
  if (t == null)
    return t;
  switch (t.key_type) {
    case "EcdsaSecp256k1":
      return { ...I(t), key_type: "EcdsaSecp256k1" };
    case "EddsaEd25519":
      return { ...D(t), key_type: "EddsaEd25519" };
    default:
      throw new Error(`No variant of SignatureWithPublicKey exists with 'key_type=${t.key_type}'`);
  }
}
function Tt(t) {
  if (t !== void 0) {
    if (t === null)
      return null;
    switch (t.key_type) {
      case "EcdsaSecp256k1":
        return ht(t);
      case "EddsaEd25519":
        return gt(t);
      default:
        throw new Error(`No variant of SignatureWithPublicKey exists with 'key_type=${t.key_type}'`);
    }
  }
}
function er(t) {
  let e = !0;
  return e = e && "key_type" in t, e = e && "key_hex" in t, e;
}
function nr(t) {
  return q(t);
}
function q(t, e) {
  return t == null ? t : {
    key_type: h(t.key_type),
    key_hex: t.key_hex
  };
}
function wt(t) {
  if (t !== void 0)
    return t === null ? null : {
      key_type: t.key_type,
      key_hex: t.key_hex
    };
}
function Jt(t) {
  return Ft(t);
}
function Ft(t, e) {
  if (t == null)
    return t;
  switch (t.key_type) {
    case "EcdsaSecp256k1":
      return { ...q(t), key_type: "EcdsaSecp256k1" };
    case "EddsaEd25519":
      return { ...V(t), key_type: "EddsaEd25519" };
    default:
      throw new Error(`No variant of PublicKey exists with 'key_type=${t.key_type}'`);
  }
}
function Rt(t) {
  if (t !== void 0) {
    if (t === null)
      return null;
    switch (t.key_type) {
      case "EcdsaSecp256k1":
        return wt(t);
      case "EddsaEd25519":
        return C(t);
      default:
        throw new Error(`No variant of PublicKey exists with 'key_type=${t.key_type}'`);
    }
  }
}
function rr(t) {
  let e = !0;
  return e = e && "version" in t, e = e && "network_id" in t, e = e && "start_epoch_inclusive" in t, e = e && "end_epoch_exclusive" in t, e = e && "nonce" in t, e = e && "notary_public_key" in t, e = e && "notary_as_signatory" in t, e = e && "cost_unit_limit" in t, e = e && "tip_percentage" in t, e;
}
function kt(t) {
  return xt(t);
}
function xt(t, e) {
  return t == null ? t : {
    version: t.version,
    network_id: t.network_id,
    start_epoch_inclusive: t.start_epoch_inclusive,
    end_epoch_exclusive: t.end_epoch_exclusive,
    nonce: t.nonce,
    notary_public_key: Jt(t.notary_public_key),
    notary_as_signatory: t.notary_as_signatory,
    cost_unit_limit: t.cost_unit_limit,
    tip_percentage: t.tip_percentage
  };
}
function Pt(t) {
  if (t !== void 0)
    return t === null ? null : {
      version: t.version,
      network_id: t.network_id,
      start_epoch_inclusive: t.start_epoch_inclusive,
      end_epoch_exclusive: t.end_epoch_exclusive,
      nonce: t.nonce,
      notary_public_key: Rt(t.notary_public_key),
      notary_as_signatory: t.notary_as_signatory,
      cost_unit_limit: t.cost_unit_limit,
      tip_percentage: t.tip_percentage
    };
}
function ir(t) {
  let e = !0;
  return e = e && "hash" in t, e = e && "header" in t, e = e && "manifest" in t, e = e && "blobs_hex" in t, e;
}
function Et(t) {
  return It(t);
}
function It(t, e) {
  return t == null ? t : {
    hash: t.hash,
    header: kt(t.header),
    manifest: t.manifest,
    blobs_hex: t.blobs_hex
  };
}
function Vt(t) {
  if (t !== void 0)
    return t === null ? null : {
      hash: t.hash,
      header: Pt(t.header),
      manifest: t.manifest,
      blobs_hex: t.blobs_hex
    };
}
function or(t) {
  let e = !0;
  return e = e && "hash" in t, e = e && "intent" in t, e = e && "intent_signatures" in t, e;
}
function Ct(t) {
  return Dt(t);
}
function Dt(t, e) {
  return t == null ? t : {
    hash: t.hash,
    intent: Et(t.intent),
    intent_signatures: t.intent_signatures.map(Ot)
  };
}
function qt(t) {
  if (t !== void 0)
    return t === null ? null : {
      hash: t.hash,
      intent: Vt(t.intent),
      intent_signatures: t.intent_signatures.map(Tt)
    };
}
function sr(t) {
  let e = !0;
  return e = e && "hash" in t, e = e && "payload_hex" in t, e = e && "signed_intent" in t, e = e && "notary_signature" in t, e;
}
function At(t) {
  return Kt(t);
}
function Kt(t, e) {
  return t == null ? t : {
    hash: t.hash,
    payload_hex: t.payload_hex,
    signed_intent: Ct(t.signed_intent),
    notary_signature: yt(t.notary_signature)
  };
}
function Mt(t) {
  if (t !== void 0)
    return t === null ? null : {
      hash: t.hash,
      payload_hex: t.payload_hex,
      signed_intent: qt(t.signed_intent),
      notary_signature: St(t.notary_signature)
    };
}
function ur(t) {
  let e = !0;
  return e = e && "loan_fully_repaid" in t, e = e && "cost_unit_limit" in t, e = e && "cost_unit_consumed" in t, e = e && "cost_unit_price_attos" in t, e = e && "tip_percentage" in t, e = e && "xrd_burned_attos" in t, e = e && "xrd_tipped_attos" in t, e;
}
function $t(t) {
  return Ut(t);
}
function Ut(t, e) {
  return t == null ? t : {
    loan_fully_repaid: t.loan_fully_repaid,
    cost_unit_limit: t.cost_unit_limit,
    cost_unit_consumed: t.cost_unit_consumed,
    cost_unit_price_attos: t.cost_unit_price_attos,
    tip_percentage: t.tip_percentage,
    xrd_burned_attos: t.xrd_burned_attos,
    xrd_tipped_attos: t.xrd_tipped_attos
  };
}
function Wt(t) {
  if (t !== void 0)
    return t === null ? null : {
      loan_fully_repaid: t.loan_fully_repaid,
      cost_unit_limit: t.cost_unit_limit,
      cost_unit_consumed: t.cost_unit_consumed,
      cost_unit_price_attos: t.cost_unit_price_attos,
      tip_percentage: t.tip_percentage,
      xrd_burned_attos: t.xrd_burned_attos,
      xrd_tipped_attos: t.xrd_tipped_attos
    };
}
function ar(t) {
  let e = !0;
  return e = e && "data_hex" in t, e = e && "data_json" in t, e;
}
function A(t) {
  return zt(t);
}
function zt(t, e) {
  return t == null ? t : {
    data_hex: t.data_hex,
    data_json: t.data_json
  };
}
function K(t) {
  if (t !== void 0)
    return t === null ? null : {
      data_hex: t.data_hex,
      data_json: t.data_json
    };
}
const cr = {
  System: "System",
  ResourceManager: "ResourceManager",
  Component: "Component",
  Package: "Package",
  Vault: "Vault",
  KeyValueStore: "KeyValueStore"
};
function u(t) {
  return Bt(t);
}
function Bt(t, e) {
  return t;
}
function dr(t) {
  return t;
}
const _r = {
  System: "System",
  ResourceManager: "ResourceManager",
  ComponentInfo: "ComponentInfo",
  ComponentState: "ComponentState",
  Package: "Package",
  Vault: "Vault",
  NonFungible: "NonFungible",
  KeyValueStoreEntry: "KeyValueStoreEntry"
};
function c(t) {
  return Ht(t);
}
function Ht(t, e) {
  return t;
}
function fr(t) {
  return t;
}
function pr(t) {
  let e = !0;
  return e = e && "entity_type" in t, e = e && "entity_address_hex" in t, e = e && "substate_type" in t, e = e && "substate_key_hex" in t, e;
}
function N(t) {
  return Gt(t);
}
function Gt(t, e) {
  return t == null ? t : {
    entity_type: u(t.entity_type),
    entity_address_hex: t.entity_address_hex,
    substate_type: c(t.substate_type),
    substate_key_hex: t.substate_key_hex
  };
}
function T(t) {
  if (t !== void 0)
    return t === null ? null : {
      entity_type: t.entity_type,
      entity_address_hex: t.entity_address_hex,
      substate_type: t.substate_type,
      substate_key_hex: t.substate_key_hex
    };
}
function lr(t) {
  let e = !0;
  return e = e && "substate_id" in t, e = e && "substate_data_hash" in t, e = e && "version" in t, e;
}
function Lt(t) {
  return Qt(t);
}
function Qt(t, e) {
  return t == null ? t : {
    substate_id: N(t.substate_id),
    substate_data_hash: t.substate_data_hash,
    version: t.version
  };
}
function Xt(t) {
  if (t !== void 0)
    return t === null ? null : {
      substate_id: T(t.substate_id),
      substate_data_hash: t.substate_data_hash,
      version: t.version
    };
}
function yr(t) {
  let e = !0;
  return e = e && "entity_type" in t, e = e && "entity_address_hex" in t, e = e && "global_address_hex" in t, e = e && "global_address" in t, e;
}
function Yt(t) {
  return Zt(t);
}
function Zt(t, e) {
  return t == null ? t : {
    entity_type: u(t.entity_type),
    entity_address_hex: t.entity_address_hex,
    global_address_hex: t.global_address_hex,
    global_address: t.global_address
  };
}
function vt(t) {
  if (t !== void 0)
    return t === null ? null : {
      entity_type: t.entity_type,
      entity_address_hex: t.entity_address_hex,
      global_address_hex: t.global_address_hex,
      global_address: t.global_address
    };
}
function mr(t) {
  let e = !0;
  return e = e && "entity_type" in t, e = e && "substate_type" in t, e = e && "package_address" in t, e = e && "blueprint_name" in t, e;
}
function Sr(t) {
  return M(t);
}
function M(t, e) {
  return t == null ? t : {
    entity_type: u(t.entity_type),
    substate_type: c(t.substate_type),
    package_address: t.package_address,
    blueprint_name: t.blueprint_name
  };
}
function jt(t) {
  if (t !== void 0)
    return t === null ? null : {
      entity_type: t.entity_type,
      substate_type: t.substate_type,
      package_address: t.package_address,
      blueprint_name: t.blueprint_name
    };
}
function hr(t) {
  let e = !0;
  return e = e && "entity_type" in t, e = e && "entity_address_hex" in t, e;
}
function g(t) {
  return te(t);
}
function te(t, e) {
  return t == null ? t : {
    entity_type: u(t.entity_type),
    entity_address_hex: t.entity_address_hex
  };
}
function O(t) {
  if (t !== void 0)
    return t === null ? null : {
      entity_type: t.entity_type,
      entity_address_hex: t.entity_address_hex
    };
}
function br(t) {
  let e = !0;
  return e = e && "struct_data" in t, e = e && "owned_entities" in t, e = e && "referenced_entities" in t, e;
}
function m(t) {
  return ee(t);
}
function ee(t, e) {
  return t == null ? t : {
    struct_data: A(t.struct_data),
    owned_entities: t.owned_entities.map(g),
    referenced_entities: t.referenced_entities.map(g)
  };
}
function S(t) {
  if (t !== void 0)
    return t === null ? null : {
      struct_data: K(t.struct_data),
      owned_entities: t.owned_entities.map(O),
      referenced_entities: t.referenced_entities.map(O)
    };
}
function gr(t) {
  let e = !0;
  return e = e && "entity_type" in t, e = e && "substate_type" in t, e = e && "data_struct" in t, e;
}
function Or(t) {
  return $(t);
}
function $(t, e) {
  return t == null ? t : {
    entity_type: u(t.entity_type),
    substate_type: c(t.substate_type),
    data_struct: m(t.data_struct)
  };
}
function ne(t) {
  if (t !== void 0)
    return t === null ? null : {
      entity_type: t.entity_type,
      substate_type: t.substate_type,
      data_struct: S(t.data_struct)
    };
}
function Nr(t) {
  let e = !0;
  return e = e && "entity_type" in t, e = e && "substate_type" in t, e = e && "key_hex" in t, e = e && "is_deleted" in t, e;
}
function Tr(t) {
  return U(t);
}
function U(t, e) {
  return t == null ? t : {
    entity_type: u(t.entity_type),
    substate_type: c(t.substate_type),
    key_hex: t.key_hex,
    is_deleted: t.is_deleted,
    data_struct: a(t, "data_struct") ? m(t.data_struct) : void 0
  };
}
function re(t) {
  if (t !== void 0)
    return t === null ? null : {
      entity_type: t.entity_type,
      substate_type: t.substate_type,
      key_hex: t.key_hex,
      is_deleted: t.is_deleted,
      data_struct: S(t.data_struct)
    };
}
function wr(t) {
  let e = !0;
  return e = e && "immutable_data" in t, e = e && "mutable_data" in t, e;
}
function W(t) {
  return ie(t);
}
function ie(t, e) {
  return t == null ? t : {
    immutable_data: m(t.immutable_data),
    mutable_data: m(t.mutable_data)
  };
}
function z(t) {
  if (t !== void 0)
    return t === null ? null : {
      immutable_data: S(t.immutable_data),
      mutable_data: S(t.mutable_data)
    };
}
function Jr(t) {
  let e = !0;
  return e = e && "entity_type" in t, e = e && "substate_type" in t, e = e && "nf_id_hex" in t, e = e && "is_deleted" in t, e;
}
function Fr(t) {
  return B(t);
}
function B(t, e) {
  return t == null ? t : {
    entity_type: u(t.entity_type),
    substate_type: c(t.substate_type),
    nf_id_hex: t.nf_id_hex,
    is_deleted: t.is_deleted,
    non_fungible_data: a(t, "non_fungible_data") ? W(t.non_fungible_data) : void 0
  };
}
function oe(t) {
  if (t !== void 0)
    return t === null ? null : {
      entity_type: t.entity_type,
      substate_type: t.substate_type,
      nf_id_hex: t.nf_id_hex,
      is_deleted: t.is_deleted,
      non_fungible_data: z(t.non_fungible_data)
    };
}
function Rr(t) {
  let e = !0;
  return e = e && "entity_type" in t, e = e && "substate_type" in t, e = e && "code_hex" in t, e;
}
function kr(t) {
  return H(t);
}
function H(t, e) {
  return t == null ? t : {
    entity_type: u(t.entity_type),
    substate_type: c(t.substate_type),
    code_hex: t.code_hex
  };
}
function se(t) {
  if (t !== void 0)
    return t === null ? null : {
      entity_type: t.entity_type,
      substate_type: t.substate_type,
      code_hex: t.code_hex
    };
}
function xr(t) {
  let e = !0;
  return e = e && "key" in t, e = e && "value" in t, e;
}
function G(t) {
  return ue(t);
}
function ue(t, e) {
  return t == null ? t : {
    key: t.key,
    value: t.value
  };
}
function L(t) {
  if (t !== void 0)
    return t === null ? null : {
      key: t.key,
      value: t.value
    };
}
const Pr = {
  Fungible: "Fungible",
  NonFungible: "NonFungible"
};
function b(t) {
  return ae(t);
}
function ae(t, e) {
  return t;
}
function Er(t) {
  return t;
}
function Ir(t) {
  let e = !0;
  return e = e && "entity_type" in t, e = e && "substate_type" in t, e = e && "resource_type" in t, e = e && "metadata" in t, e = e && "total_supply_attos" in t, e;
}
function Vr(t) {
  return Q(t);
}
function Q(t, e) {
  return t == null ? t : {
    entity_type: u(t.entity_type),
    substate_type: c(t.substate_type),
    resource_type: b(t.resource_type),
    fungible_divisibility: a(t, "fungible_divisibility") ? t.fungible_divisibility : void 0,
    metadata: t.metadata.map(G),
    total_supply_attos: t.total_supply_attos
  };
}
function ce(t) {
  if (t !== void 0)
    return t === null ? null : {
      entity_type: t.entity_type,
      substate_type: t.substate_type,
      resource_type: t.resource_type,
      fungible_divisibility: t.fungible_divisibility,
      metadata: t.metadata.map(L),
      total_supply_attos: t.total_supply_attos
    };
}
function Cr(t) {
  let e = !0;
  return e = e && "entity_type" in t, e = e && "substate_type" in t, e = e && "epoch" in t, e;
}
function Dr(t) {
  return X(t);
}
function X(t, e) {
  return t == null ? t : {
    entity_type: u(t.entity_type),
    substate_type: c(t.substate_type),
    epoch: t.epoch
  };
}
function de(t) {
  if (t !== void 0)
    return t === null ? null : {
      entity_type: t.entity_type,
      substate_type: t.substate_type,
      epoch: t.epoch
    };
}
function qr(t) {
  let e = !0;
  return e = e && "resource_type" in t, e = e && "resource_address" in t, e = e && "amount_attos" in t, e;
}
function Ar(t) {
  return Y(t);
}
function Y(t, e) {
  return t == null ? t : {
    resource_type: b(t.resource_type),
    resource_address: t.resource_address,
    amount_attos: t.amount_attos
  };
}
function _e(t) {
  if (t !== void 0)
    return t === null ? null : {
      resource_type: t.resource_type,
      resource_address: t.resource_address,
      amount_attos: t.amount_attos
    };
}
function Kr(t) {
  let e = !0;
  return e = e && "resource_type" in t, e = e && "resource_address" in t, e = e && "nf_ids_hex" in t, e;
}
function Mr(t) {
  return Z(t);
}
function Z(t, e) {
  return t == null ? t : {
    resource_type: b(t.resource_type),
    resource_address: t.resource_address,
    nf_ids_hex: t.nf_ids_hex
  };
}
function fe(t) {
  if (t !== void 0)
    return t === null ? null : {
      resource_type: t.resource_type,
      resource_address: t.resource_address,
      nf_ids_hex: t.nf_ids_hex
    };
}
function v(t) {
  return pe(t);
}
function pe(t, e) {
  if (t == null)
    return t;
  switch (t.resource_type) {
    case "Fungible":
      return { ...Y(t), resource_type: "Fungible" };
    case "NonFungible":
      return { ...Z(t), resource_type: "NonFungible" };
    default:
      throw new Error(`No variant of ResourceAmount exists with 'resource_type=${t.resource_type}'`);
  }
}
function j(t) {
  if (t !== void 0) {
    if (t === null)
      return null;
    switch (t.resource_type) {
      case "Fungible":
        return _e(t);
      case "NonFungible":
        return fe(t);
      default:
        throw new Error(`No variant of ResourceAmount exists with 'resource_type=${t.resource_type}'`);
    }
  }
}
function $r(t) {
  let e = !0;
  return e = e && "entity_type" in t, e = e && "substate_type" in t, e = e && "resource_amount" in t, e;
}
function Ur(t) {
  return tt(t);
}
function tt(t, e) {
  return t == null ? t : {
    entity_type: u(t.entity_type),
    substate_type: c(t.substate_type),
    resource_amount: v(t.resource_amount)
  };
}
function le(t) {
  if (t !== void 0)
    return t === null ? null : {
      entity_type: t.entity_type,
      substate_type: t.substate_type,
      resource_amount: j(t.resource_amount)
    };
}
function f(t) {
  return ye(t);
}
function ye(t, e) {
  if (t == null)
    return t;
  switch (t.substate_type) {
    case "ComponentInfo":
      return { ...M(t), substate_type: "ComponentInfo" };
    case "ComponentState":
      return { ...$(t), substate_type: "ComponentState" };
    case "KeyValueStoreEntry":
      return { ...U(t), substate_type: "KeyValueStoreEntry" };
    case "NonFungible":
      return { ...B(t), substate_type: "NonFungible" };
    case "Package":
      return { ...H(t), substate_type: "Package" };
    case "ResourceManager":
      return { ...Q(t), substate_type: "ResourceManager" };
    case "System":
      return { ...X(t), substate_type: "System" };
    case "Vault":
      return { ...tt(t), substate_type: "Vault" };
    default:
      throw new Error(`No variant of Substate exists with 'substate_type=${t.substate_type}'`);
  }
}
function p(t) {
  if (t !== void 0) {
    if (t === null)
      return null;
    switch (t.substate_type) {
      case "ComponentInfo":
        return jt(t);
      case "ComponentState":
        return ne(t);
      case "KeyValueStoreEntry":
        return re(t);
      case "NonFungible":
        return oe(t);
      case "Package":
        return se(t);
      case "ResourceManager":
        return ce(t);
      case "System":
        return de(t);
      case "Vault":
        return le(t);
      default:
        throw new Error(`No variant of Substate exists with 'substate_type=${t.substate_type}'`);
    }
  }
}
function Wr(t) {
  let e = !0;
  return e = e && "substate_id" in t, e = e && "version" in t, e = e && "substate_hex" in t, e = e && "substate_data_hash" in t, e = e && "substate_data" in t, e;
}
function me(t) {
  return Se(t);
}
function Se(t, e) {
  return t == null ? t : {
    substate_id: N(t.substate_id),
    version: t.version,
    substate_hex: t.substate_hex,
    substate_data_hash: t.substate_data_hash,
    substate_data: f(t.substate_data)
  };
}
function he(t) {
  if (t !== void 0)
    return t === null ? null : {
      substate_id: T(t.substate_id),
      version: t.version,
      substate_hex: t.substate_hex,
      substate_data_hash: t.substate_data_hash,
      substate_data: p(t.substate_data)
    };
}
function zr(t) {
  let e = !0;
  return e = e && "down_virtual_substates" in t, e = e && "up_substates" in t, e = e && "down_substates" in t, e = e && "new_global_entities" in t, e;
}
function be(t) {
  return ge(t);
}
function ge(t, e) {
  return t == null ? t : {
    down_virtual_substates: t.down_virtual_substates.map(N),
    up_substates: t.up_substates.map(me),
    down_substates: t.down_substates.map(Lt),
    new_global_entities: t.new_global_entities.map(Yt)
  };
}
function Oe(t) {
  if (t !== void 0)
    return t === null ? null : {
      down_virtual_substates: t.down_virtual_substates.map(T),
      up_substates: t.up_substates.map(he),
      down_substates: t.down_substates.map(Xt),
      new_global_entities: t.new_global_entities.map(vt)
    };
}
const Br = {
  Succeeded: "Succeeded",
  Failed: "Failed",
  Rejected: "Rejected"
};
function Ne(t) {
  return Te(t);
}
function Te(t, e) {
  return t;
}
function Hr(t) {
  return t;
}
function Gr(t) {
  let e = !0;
  return e = e && "status" in t, e = e && "fee_summary" in t, e = e && "state_updates" in t, e;
}
function we(t) {
  return Je(t);
}
function Je(t, e) {
  return t == null ? t : {
    status: Ne(t.status),
    fee_summary: $t(t.fee_summary),
    state_updates: be(t.state_updates),
    output: a(t, "output") ? t.output.map(A) : void 0,
    error_message: a(t, "error_message") ? t.error_message : void 0
  };
}
function Fe(t) {
  if (t !== void 0)
    return t === null ? null : {
      status: t.status,
      fee_summary: Wt(t.fee_summary),
      state_updates: Oe(t.state_updates),
      output: t.output === void 0 ? void 0 : t.output.map(K),
      error_message: t.error_message
    };
}
function Lr(t) {
  let e = !0;
  return e = e && "state_version" in t, e = e && "receipt" in t, e;
}
function Re(t) {
  return ke(t);
}
function ke(t, e) {
  return t == null ? t : {
    state_version: t.state_version,
    notarized_transaction: a(t, "notarized_transaction") ? At(t.notarized_transaction) : void 0,
    receipt: we(t.receipt)
  };
}
function xe(t) {
  if (t !== void 0)
    return t === null ? null : {
      state_version: t.state_version,
      notarized_transaction: Mt(t.notarized_transaction),
      receipt: Fe(t.receipt)
    };
}
function Qr(t) {
  let e = !0;
  return e = e && "package_address" in t, e = e && "blueprint_name" in t, e;
}
function Xr(t) {
  return Pe(t);
}
function Pe(t, e) {
  return t == null ? t : {
    package_address: t.package_address,
    blueprint_name: t.blueprint_name
  };
}
function Yr(t) {
  if (t !== void 0)
    return t === null ? null : {
      package_address: t.package_address,
      blueprint_name: t.blueprint_name
    };
}
function Zr(t) {
  let e = !0;
  return e = e && "data_struct" in t, e;
}
function vr(t) {
  return Ee(t);
}
function Ee(t, e) {
  return t == null ? t : {
    data_struct: m(t.data_struct)
  };
}
function jr(t) {
  if (t !== void 0)
    return t === null ? null : {
      data_struct: S(t.data_struct)
    };
}
function ti(t) {
  let e = !0;
  return e = e && "code" in t, e = e && "message" in t, e;
}
function ei(t) {
  return Ie(t);
}
function Ie(t, e) {
  return t == null ? t : {
    code: t.code,
    message: t.message,
    trace_id: a(t, "trace_id") ? t.trace_id : void 0
  };
}
function ni(t) {
  if (t !== void 0)
    return t === null ? null : {
      code: t.code,
      message: t.message,
      trace_id: t.trace_id
    };
}
function ri(t) {
  let e = !0;
  return e = e && "amount_attos" in t, e;
}
function ii(t) {
  return Ve(t);
}
function Ve(t, e) {
  return t == null ? t : {
    amount_attos: t.amount_attos
  };
}
function oi(t) {
  if (t !== void 0)
    return t === null ? null : {
      amount_attos: t.amount_attos
    };
}
function si(t) {
  let e = !0;
  return e = e && "key_hex" in t, e = e && "is_deleted" in t, e;
}
function ui(t) {
  return Ce(t);
}
function Ce(t, e) {
  return t == null ? t : {
    key_hex: t.key_hex,
    is_deleted: t.is_deleted,
    data_struct: a(t, "data_struct") ? m(t.data_struct) : void 0
  };
}
function ai(t) {
  if (t !== void 0)
    return t === null ? null : {
      key_hex: t.key_hex,
      is_deleted: t.is_deleted,
      data_struct: S(t.data_struct)
    };
}
function ci(t) {
  let e = !0;
  return e = e && "nf_ids_hex" in t, e;
}
function di(t) {
  return De(t);
}
function De(t, e) {
  return t == null ? t : {
    nf_ids_hex: t.nf_ids_hex
  };
}
function _i(t) {
  if (t !== void 0)
    return t === null ? null : {
      nf_ids_hex: t.nf_ids_hex
    };
}
function fi(t) {
  let e = !0;
  return e = e && "nf_id_hex" in t, e = e && "is_deleted" in t, e;
}
function pi(t) {
  return qe(t);
}
function qe(t, e) {
  return t == null ? t : {
    nf_id_hex: t.nf_id_hex,
    is_deleted: t.is_deleted,
    non_fungible_data: a(t, "non_fungible_data") ? W(t.non_fungible_data) : void 0
  };
}
function li(t) {
  if (t !== void 0)
    return t === null ? null : {
      nf_id_hex: t.nf_id_hex,
      is_deleted: t.is_deleted,
      non_fungible_data: z(t.non_fungible_data)
    };
}
function yi(t) {
  let e = !0;
  return e = e && "code_hex" in t, e;
}
function mi(t) {
  return Ae(t);
}
function Ae(t, e) {
  return t == null ? t : {
    code_hex: t.code_hex
  };
}
function Si(t) {
  if (t !== void 0)
    return t === null ? null : {
      code_hex: t.code_hex
    };
}
function hi(t) {
  let e = !0;
  return e = e && "resource_type" in t, e = e && "resource_address" in t, e;
}
function bi(t) {
  return Ke(t);
}
function Ke(t, e) {
  return t == null ? t : {
    resource_type: b(t.resource_type),
    resource_address: t.resource_address
  };
}
function gi(t) {
  if (t !== void 0)
    return t === null ? null : {
      resource_type: t.resource_type,
      resource_address: t.resource_address
    };
}
function Oi(t) {
  let e = !0;
  return e = e && "resource_address" in t, e = e && "component_address" in t, e = e && "vault_entity_id" in t, e = e && "amount_attos" in t, e;
}
function Ni(t) {
  return Me(t);
}
function Me(t, e) {
  return t == null ? t : {
    resource_address: t.resource_address,
    component_address: t.component_address,
    vault_entity_id: g(t.vault_entity_id),
    amount_attos: t.amount_attos
  };
}
function Ti(t) {
  if (t !== void 0)
    return t === null ? null : {
      resource_address: t.resource_address,
      component_address: t.component_address,
      vault_entity_id: O(t.vault_entity_id),
      amount_attos: t.amount_attos
    };
}
function wi(t) {
  let e = !0;
  return e = e && "resource_type" in t, e = e && "metadata" in t, e = e && "total_supply_attos" in t, e;
}
function Ji(t) {
  return $e(t);
}
function $e(t, e) {
  return t == null ? t : {
    resource_type: b(t.resource_type),
    fungible_divisibility: a(t, "fungible_divisibility") ? t.fungible_divisibility : void 0,
    metadata: t.metadata.map(G),
    total_supply_attos: t.total_supply_attos
  };
}
function Fi(t) {
  if (t !== void 0)
    return t === null ? null : {
      resource_type: t.resource_type,
      fungible_divisibility: t.fungible_divisibility,
      metadata: t.metadata.map(L),
      total_supply_attos: t.total_supply_attos
    };
}
function Ri(t) {
  let e = !0;
  return e = e && "entity_type" in t, e = e && "substate_type" in t, e;
}
function ki(t) {
  return Ue(t);
}
function Ue(t, e) {
  return t == null ? t : {
    entity_type: u(t.entity_type),
    substate_type: c(t.substate_type)
  };
}
function xi(t) {
  if (t !== void 0)
    return t === null ? null : {
      entity_type: t.entity_type,
      substate_type: t.substate_type
    };
}
function Pi(t) {
  let e = !0;
  return e = e && "epoch" in t, e;
}
function Ei(t) {
  return We(t);
}
function We(t, e) {
  return t == null ? t : {
    epoch: t.epoch
  };
}
function Ii(t) {
  if (t !== void 0)
    return t === null ? null : {
      epoch: t.epoch
    };
}
function Vi(t) {
  let e = !0;
  return e = e && "intent_hash" in t, e = e && "signatures_hash" in t, e = e && "payload_hash" in t, e;
}
function Ci(t) {
  return ze(t);
}
function ze(t, e) {
  return t == null ? t : {
    intent_hash: t.intent_hash,
    signatures_hash: t.signatures_hash,
    payload_hash: t.payload_hash
  };
}
function Di(t) {
  if (t !== void 0)
    return t === null ? null : {
      intent_hash: t.intent_hash,
      signatures_hash: t.signatures_hash,
      payload_hash: t.payload_hash
    };
}
function qi(t) {
  let e = !0;
  return e = e && "intent_hash" in t, e;
}
function Ai(t) {
  return Be(t);
}
function Be(t, e) {
  return t == null ? t : {
    intent_hash: t.intent_hash
  };
}
function He(t) {
  if (t !== void 0)
    return t === null ? null : {
      intent_hash: t.intent_hash
    };
}
function Ki(t) {
  let e = !0;
  return e = e && "committed" in t, e;
}
function Ge(t) {
  return Le(t);
}
function Le(t, e) {
  return t == null ? t : {
    committed: Re(t.committed)
  };
}
function Mi(t) {
  if (t !== void 0)
    return t === null ? null : {
      committed: xe(t.committed)
    };
}
function $i(t) {
  let e = !0;
  return e = e && "core_version" in t, e = e && "api_version" in t, e;
}
function Qe(t) {
  return Xe(t);
}
function Xe(t, e) {
  return t == null ? t : {
    core_version: t.core_version,
    api_version: t.api_version
  };
}
function Ye(t) {
  if (t !== void 0)
    return t === null ? null : {
      core_version: t.core_version,
      api_version: t.api_version
    };
}
function Ui(t) {
  let e = !0;
  return e = e && "account_package" in t, e = e && "faucet" in t, e = e && "ecdsa_secp256k1" in t, e = e && "eddsa_ed25519" in t, e = e && "xrd" in t, e;
}
function Ze(t) {
  return ve(t);
}
function ve(t, e) {
  return t == null ? t : {
    account_package: t.account_package,
    faucet: t.faucet,
    ecdsa_secp256k1: t.ecdsa_secp256k1,
    eddsa_ed25519: t.eddsa_ed25519,
    xrd: t.xrd
  };
}
function je(t) {
  if (t !== void 0)
    return t === null ? null : {
      account_package: t.account_package,
      faucet: t.faucet,
      ecdsa_secp256k1: t.ecdsa_secp256k1,
      eddsa_ed25519: t.eddsa_ed25519,
      xrd: t.xrd
    };
}
function Wi(t) {
  let e = !0;
  return e = e && "version" in t, e = e && "network" in t, e = e && "network_hrp_suffix" in t, e = e && "well_known_addresses" in t, e;
}
function tn(t) {
  return en(t);
}
function en(t, e) {
  return t == null ? t : {
    version: Qe(t.version),
    network: t.network,
    network_hrp_suffix: t.network_hrp_suffix,
    well_known_addresses: Ze(t.well_known_addresses)
  };
}
function zi(t) {
  if (t !== void 0)
    return t === null ? null : {
      version: Ye(t.version),
      network: t.network,
      network_hrp_suffix: t.network_hrp_suffix,
      well_known_addresses: je(t.well_known_addresses)
    };
}
function Bi(t) {
  let e = !0;
  return e = e && "parent" in t, e = e && "entity_id" in t, e = e && "depth" in t, e;
}
function nn(t) {
  return rn(t);
}
function rn(t, e) {
  return t == null ? t : {
    parent: N(t.parent),
    entity_id: g(t.entity_id),
    depth: t.depth
  };
}
function on(t) {
  if (t !== void 0)
    return t === null ? null : {
      parent: T(t.parent),
      entity_id: O(t.entity_id),
      depth: t.depth
    };
}
function Hi(t) {
  let e = !0;
  return e = e && "component_address" in t, e;
}
function Gi(t) {
  return sn(t);
}
function sn(t, e) {
  return t == null ? t : {
    component_address: t.component_address
  };
}
function un(t) {
  if (t !== void 0)
    return t === null ? null : {
      component_address: t.component_address
    };
}
function Li(t) {
  let e = !0;
  return e = e && "info" in t, e = e && "state" in t, e = e && "owned_vaults" in t, e = e && "descendent_ids" in t, e;
}
function an(t) {
  return cn(t);
}
function cn(t, e) {
  return t == null ? t : {
    info: f(t.info),
    state: f(t.state),
    owned_vaults: t.owned_vaults.map(f),
    descendent_ids: t.descendent_ids.map(nn)
  };
}
function Qi(t) {
  if (t !== void 0)
    return t === null ? null : {
      info: p(t.info),
      state: p(t.state),
      owned_vaults: t.owned_vaults.map(p),
      descendent_ids: t.descendent_ids.map(on)
    };
}
function Xi(t) {
  let e = !0;
  return e = e && "epoch" in t, e;
}
function dn(t) {
  return _n(t);
}
function _n(t, e) {
  return t == null ? t : {
    epoch: t.epoch
  };
}
function Yi(t) {
  if (t !== void 0)
    return t === null ? null : {
      epoch: t.epoch
    };
}
function Zi(t) {
  let e = !0;
  return e = e && "resource_address" in t, e = e && "non_fungible_id_hex" in t, e;
}
function vi(t) {
  return fn(t);
}
function fn(t, e) {
  return t == null ? t : {
    resource_address: t.resource_address,
    non_fungible_id_hex: t.non_fungible_id_hex
  };
}
function pn(t) {
  if (t !== void 0)
    return t === null ? null : {
      resource_address: t.resource_address,
      non_fungible_id_hex: t.non_fungible_id_hex
    };
}
function ji(t) {
  let e = !0;
  return e = e && "non_fungible" in t, e;
}
function ln(t) {
  return yn(t);
}
function yn(t, e) {
  return t == null ? t : {
    non_fungible: f(t.non_fungible)
  };
}
function to(t) {
  if (t !== void 0)
    return t === null ? null : {
      non_fungible: p(t.non_fungible)
    };
}
function eo(t) {
  let e = !0;
  return e = e && "package_address" in t, e;
}
function no(t) {
  return mn(t);
}
function mn(t, e) {
  return t == null ? t : {
    package_address: t.package_address
  };
}
function Sn(t) {
  if (t !== void 0)
    return t === null ? null : {
      package_address: t.package_address
    };
}
function ro(t) {
  let e = !0;
  return e = e && "_package" in t, e;
}
function hn(t) {
  return bn(t);
}
function bn(t, e) {
  return t == null ? t : {
    _package: f(t.package)
  };
}
function io(t) {
  if (t !== void 0)
    return t === null ? null : {
      package: p(t._package)
    };
}
function oo(t) {
  let e = !0;
  return e = e && "resource_address" in t, e;
}
function so(t) {
  return gn(t);
}
function gn(t, e) {
  return t == null ? t : {
    resource_address: t.resource_address
  };
}
function On(t) {
  if (t !== void 0)
    return t === null ? null : {
      resource_address: t.resource_address
    };
}
function uo(t) {
  let e = !0;
  return e = e && "manager" in t, e;
}
function Nn(t) {
  return Tn(t);
}
function Tn(t, e) {
  return t == null ? t : {
    manager: f(t.manager)
  };
}
function ao(t) {
  if (t !== void 0)
    return t === null ? null : {
      manager: p(t.manager)
    };
}
const co = {
  CommittedSuccess: "CommittedSuccess",
  CommittedFailure: "CommittedFailure",
  InMempool: "InMempool",
  Rejected: "Rejected"
};
function _o(t) {
  let e = !0;
  return e = e && "payload_hash" in t, e = e && "status" in t, e;
}
function wn(t) {
  return Jn(t);
}
function Jn(t, e) {
  return t == null ? t : {
    payload_hash: t.payload_hash,
    status: t.status,
    error_message: a(t, "error_message") ? t.error_message : void 0
  };
}
function Fn(t) {
  if (t !== void 0)
    return t === null ? null : {
      payload_hash: t.payload_hash,
      status: t.status,
      error_message: t.error_message
    };
}
function fo(t) {
  let e = !0;
  return e = e && "intent_hash" in t, e;
}
function po(t) {
  return Rn(t);
}
function Rn(t, e) {
  return t == null ? t : {
    intent_hash: t.intent_hash
  };
}
function kn(t) {
  if (t !== void 0)
    return t === null ? null : {
      intent_hash: t.intent_hash
    };
}
const lo = {
  CommittedSuccess: "CommittedSuccess",
  CommittedFailure: "CommittedFailure",
  InMempool: "InMempool",
  Rejected: "Rejected",
  Unknown: "Unknown"
};
function yo(t) {
  let e = !0;
  return e = e && "intent_status" in t, e = e && "known_payloads" in t, e;
}
function xn(t) {
  return Pn(t);
}
function Pn(t, e) {
  return t == null ? t : {
    intent_status: t.intent_status,
    known_payloads: t.known_payloads.map(wn)
  };
}
function mo(t) {
  if (t !== void 0)
    return t === null ? null : {
      intent_status: t.intent_status,
      known_payloads: t.known_payloads.map(Fn)
    };
}
function So(t) {
  let e = !0;
  return e = e && "notarized_transaction_hex" in t, e;
}
function ho(t) {
  return En(t);
}
function En(t, e) {
  return t == null ? t : {
    notarized_transaction_hex: t.notarized_transaction_hex
  };
}
function In(t) {
  if (t !== void 0)
    return t === null ? null : {
      notarized_transaction_hex: t.notarized_transaction_hex
    };
}
function bo(t) {
  let e = !0;
  return e = e && "duplicate" in t, e;
}
function Vn(t) {
  return Cn(t);
}
function Cn(t, e) {
  return t == null ? t : {
    duplicate: t.duplicate
  };
}
function go(t) {
  if (t !== void 0)
    return t === null ? null : {
      duplicate: t.duplicate
    };
}
function Oo(t) {
  let e = !0;
  return e = e && "resource_amount" in t, e;
}
function No(t) {
  return Dn(t);
}
function Dn(t, e) {
  return t == null ? t : {
    resource_amount: v(t.resource_amount)
  };
}
function To(t) {
  if (t !== void 0)
    return t === null ? null : {
      resource_amount: j(t.resource_amount)
    };
}
class wo extends w {
  async stateComponentPostRaw(e, n) {
    if (e.v0StateComponentRequest === null || e.v0StateComponentRequest === void 0)
      throw new _("v0StateComponentRequest", "Required parameter requestParameters.v0StateComponentRequest was null or undefined when calling stateComponentPost.");
    const r = {}, i = {};
    i["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/state/component",
      method: "POST",
      headers: i,
      query: r,
      body: un(e.v0StateComponentRequest)
    }, n);
    return new d(o, (s) => an(s));
  }
  async stateComponentPost(e, n) {
    return await (await this.stateComponentPostRaw(e, n)).value();
  }
  async stateEpochPostRaw(e) {
    const n = {}, r = {}, i = await this.request({
      path: "/state/epoch",
      method: "POST",
      headers: r,
      query: n
    }, e);
    return new d(i, (o) => dn(o));
  }
  async stateEpochPost(e) {
    return await (await this.stateEpochPostRaw(e)).value();
  }
  async stateNonFungiblePostRaw(e, n) {
    if (e.v0StateNonFungibleRequest === null || e.v0StateNonFungibleRequest === void 0)
      throw new _("v0StateNonFungibleRequest", "Required parameter requestParameters.v0StateNonFungibleRequest was null or undefined when calling stateNonFungiblePost.");
    const r = {}, i = {};
    i["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/state/non-fungible",
      method: "POST",
      headers: i,
      query: r,
      body: pn(e.v0StateNonFungibleRequest)
    }, n);
    return new d(o, (s) => ln(s));
  }
  async stateNonFungiblePost(e, n) {
    return await (await this.stateNonFungiblePostRaw(e, n)).value();
  }
  async statePackagePostRaw(e, n) {
    if (e.v0StatePackageRequest === null || e.v0StatePackageRequest === void 0)
      throw new _("v0StatePackageRequest", "Required parameter requestParameters.v0StatePackageRequest was null or undefined when calling statePackagePost.");
    const r = {}, i = {};
    i["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/state/package",
      method: "POST",
      headers: i,
      query: r,
      body: Sn(e.v0StatePackageRequest)
    }, n);
    return new d(o, (s) => hn(s));
  }
  async statePackagePost(e, n) {
    return await (await this.statePackagePostRaw(e, n)).value();
  }
  async stateResourcePostRaw(e, n) {
    if (e.v0StateResourceRequest === null || e.v0StateResourceRequest === void 0)
      throw new _("v0StateResourceRequest", "Required parameter requestParameters.v0StateResourceRequest was null or undefined when calling stateResourcePost.");
    const r = {}, i = {};
    i["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/state/resource",
      method: "POST",
      headers: i,
      query: r,
      body: On(e.v0StateResourceRequest)
    }, n);
    return new d(o, (s) => Nn(s));
  }
  async stateResourcePost(e, n) {
    return await (await this.stateResourcePostRaw(e, n)).value();
  }
}
class Jo extends w {
  async statusNetworkConfigurationPostRaw(e) {
    const n = {}, r = {}, i = await this.request({
      path: "/status/network-configuration",
      method: "POST",
      headers: r,
      query: n
    }, e);
    return new d(i, (o) => tn(o));
  }
  async statusNetworkConfigurationPost(e) {
    return await (await this.statusNetworkConfigurationPostRaw(e)).value();
  }
}
class Fo extends w {
  async transactionReceiptPostRaw(e, n) {
    if (e.v0CommittedTransactionRequest === null || e.v0CommittedTransactionRequest === void 0)
      throw new _("v0CommittedTransactionRequest", "Required parameter requestParameters.v0CommittedTransactionRequest was null or undefined when calling transactionReceiptPost.");
    const r = {}, i = {};
    i["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/transaction/receipt",
      method: "POST",
      headers: i,
      query: r,
      body: He(e.v0CommittedTransactionRequest)
    }, n);
    return new d(o, (s) => Ge(s));
  }
  async transactionReceiptPost(e, n) {
    return await (await this.transactionReceiptPostRaw(e, n)).value();
  }
  async transactionStatusPostRaw(e, n) {
    if (e.v0TransactionStatusRequest === null || e.v0TransactionStatusRequest === void 0)
      throw new _("v0TransactionStatusRequest", "Required parameter requestParameters.v0TransactionStatusRequest was null or undefined when calling transactionStatusPost.");
    const r = {}, i = {};
    i["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/transaction/status",
      method: "POST",
      headers: i,
      query: r,
      body: kn(e.v0TransactionStatusRequest)
    }, n);
    return new d(o, (s) => xn(s));
  }
  async transactionStatusPost(e, n) {
    return await (await this.transactionStatusPostRaw(e, n)).value();
  }
  async transactionSubmitPostRaw(e, n) {
    if (e.v0TransactionSubmitRequest === null || e.v0TransactionSubmitRequest === void 0)
      throw new _("v0TransactionSubmitRequest", "Required parameter requestParameters.v0TransactionSubmitRequest was null or undefined when calling transactionSubmitPost.");
    const r = {}, i = {};
    i["Content-Type"] = "application/json";
    const o = await this.request({
      path: "/transaction/submit",
      method: "POST",
      headers: i,
      query: r,
      body: In(e.v0TransactionSubmitRequest)
    }, n);
    return new d(o, (s) => Vn(s));
  }
  async transactionSubmitPost(e, n) {
    return await (await this.transactionSubmitPostRaw(e, n)).value();
  }
}
export {
  it as BASE_PATH,
  w as BaseAPI,
  Un as BlobApiResponse,
  An as COLLECTION_FORMATS,
  Bn as CommittedStateIdentifierFromJSON,
  _t as CommittedStateIdentifierFromJSONTyped,
  Hn as CommittedStateIdentifierToJSON,
  Re as CommittedTransactionFromJSON,
  ke as CommittedTransactionFromJSONTyped,
  xe as CommittedTransactionToJSON,
  Xr as ComponentInfoSubstateAllOfFromJSON,
  Pe as ComponentInfoSubstateAllOfFromJSONTyped,
  Yr as ComponentInfoSubstateAllOfToJSON,
  Sr as ComponentInfoSubstateFromJSON,
  M as ComponentInfoSubstateFromJSONTyped,
  jt as ComponentInfoSubstateToJSON,
  vr as ComponentStateSubstateAllOfFromJSON,
  Ee as ComponentStateSubstateAllOfFromJSONTyped,
  jr as ComponentStateSubstateAllOfToJSON,
  Or as ComponentStateSubstateFromJSON,
  $ as ComponentStateSubstateFromJSONTyped,
  ne as ComponentStateSubstateToJSON,
  ot as Configuration,
  m as DataStructFromJSON,
  ee as DataStructFromJSONTyped,
  S as DataStructToJSON,
  st as DefaultConfig,
  Lt as DownSubstateFromJSON,
  Qt as DownSubstateFromJSONTyped,
  Xt as DownSubstateToJSON,
  nr as EcdsaSecp256k1PublicKeyFromJSON,
  q as EcdsaSecp256k1PublicKeyFromJSONTyped,
  wt as EcdsaSecp256k1PublicKeyToJSON,
  pt as EcdsaSecp256k1SignatureFromJSON,
  k as EcdsaSecp256k1SignatureFromJSONTyped,
  x as EcdsaSecp256k1SignatureToJSON,
  Zn as EcdsaSecp256k1SignatureWithPublicKeyFromJSON,
  I as EcdsaSecp256k1SignatureWithPublicKeyFromJSONTyped,
  ht as EcdsaSecp256k1SignatureWithPublicKeyToJSON,
  bt as EddsaEd25519PublicKeyFromJSON,
  V as EddsaEd25519PublicKeyFromJSONTyped,
  C as EddsaEd25519PublicKeyToJSON,
  lt as EddsaEd25519SignatureFromJSON,
  P as EddsaEd25519SignatureFromJSONTyped,
  E as EddsaEd25519SignatureToJSON,
  tr as EddsaEd25519SignatureWithPublicKeyFromJSON,
  D as EddsaEd25519SignatureWithPublicKeyFromJSONTyped,
  gt as EddsaEd25519SignatureWithPublicKeyToJSON,
  g as EntityIdFromJSON,
  te as EntityIdFromJSONTyped,
  O as EntityIdToJSON,
  cr as EntityType,
  u as EntityTypeFromJSON,
  Bt as EntityTypeFromJSONTyped,
  dr as EntityTypeToJSON,
  ei as ErrorResponseFromJSON,
  Ie as ErrorResponseFromJSONTyped,
  ni as ErrorResponseToJSON,
  $t as FeeSummaryFromJSON,
  Ut as FeeSummaryFromJSONTyped,
  Wt as FeeSummaryToJSON,
  dt as FetchError,
  ii as FungibleResourceAmountAllOfFromJSON,
  Ve as FungibleResourceAmountAllOfFromJSONTyped,
  oi as FungibleResourceAmountAllOfToJSON,
  Ar as FungibleResourceAmountFromJSON,
  Y as FungibleResourceAmountFromJSONTyped,
  _e as FungibleResourceAmountToJSON,
  Yt as GlobalEntityIdFromJSON,
  Zt as GlobalEntityIdFromJSONTyped,
  vt as GlobalEntityIdToJSON,
  d as JSONApiResponse,
  ui as KeyValueStoreEntrySubstateAllOfFromJSON,
  Ce as KeyValueStoreEntrySubstateAllOfFromJSONTyped,
  ai as KeyValueStoreEntrySubstateAllOfToJSON,
  Tr as KeyValueStoreEntrySubstateFromJSON,
  U as KeyValueStoreEntrySubstateFromJSONTyped,
  re as KeyValueStoreEntrySubstateToJSON,
  W as NonFungibleDataFromJSON,
  ie as NonFungibleDataFromJSONTyped,
  z as NonFungibleDataToJSON,
  di as NonFungibleResourceAmountAllOfFromJSON,
  De as NonFungibleResourceAmountAllOfFromJSONTyped,
  _i as NonFungibleResourceAmountAllOfToJSON,
  Mr as NonFungibleResourceAmountFromJSON,
  Z as NonFungibleResourceAmountFromJSONTyped,
  fe as NonFungibleResourceAmountToJSON,
  pi as NonFungibleSubstateAllOfFromJSON,
  qe as NonFungibleSubstateAllOfFromJSONTyped,
  li as NonFungibleSubstateAllOfToJSON,
  Fr as NonFungibleSubstateFromJSON,
  B as NonFungibleSubstateFromJSONTyped,
  oe as NonFungibleSubstateToJSON,
  At as NotarizedTransactionFromJSON,
  Kt as NotarizedTransactionFromJSONTyped,
  Mt as NotarizedTransactionToJSON,
  mi as PackageSubstateAllOfFromJSON,
  Ae as PackageSubstateAllOfFromJSONTyped,
  Si as PackageSubstateAllOfToJSON,
  kr as PackageSubstateFromJSON,
  H as PackageSubstateFromJSONTyped,
  se as PackageSubstateToJSON,
  Jt as PublicKeyFromJSON,
  Ft as PublicKeyFromJSONTyped,
  Rt as PublicKeyToJSON,
  Gn as PublicKeyType,
  h as PublicKeyTypeFromJSON,
  ft as PublicKeyTypeFromJSONTyped,
  Ln as PublicKeyTypeToJSON,
  _ as RequiredError,
  bi as ResourceAmountBaseFromJSON,
  Ke as ResourceAmountBaseFromJSONTyped,
  gi as ResourceAmountBaseToJSON,
  v as ResourceAmountFromJSON,
  pe as ResourceAmountFromJSONTyped,
  j as ResourceAmountToJSON,
  Ni as ResourceChangeFromJSON,
  Me as ResourceChangeFromJSONTyped,
  Ti as ResourceChangeToJSON,
  Ji as ResourceManagerSubstateAllOfFromJSON,
  $e as ResourceManagerSubstateAllOfFromJSONTyped,
  G as ResourceManagerSubstateAllOfMetadataFromJSON,
  ue as ResourceManagerSubstateAllOfMetadataFromJSONTyped,
  L as ResourceManagerSubstateAllOfMetadataToJSON,
  Fi as ResourceManagerSubstateAllOfToJSON,
  Vr as ResourceManagerSubstateFromJSON,
  Q as ResourceManagerSubstateFromJSONTyped,
  ce as ResourceManagerSubstateToJSON,
  Pr as ResourceType,
  b as ResourceTypeFromJSON,
  ae as ResourceTypeFromJSONTyped,
  Er as ResourceTypeToJSON,
  ct as ResponseError,
  A as SborDataFromJSON,
  zt as SborDataFromJSONTyped,
  K as SborDataToJSON,
  yt as SignatureFromJSON,
  mt as SignatureFromJSONTyped,
  St as SignatureToJSON,
  Ot as SignatureWithPublicKeyFromJSON,
  Nt as SignatureWithPublicKeyFromJSONTyped,
  Tt as SignatureWithPublicKeyToJSON,
  Ct as SignedTransactionIntentFromJSON,
  Dt as SignedTransactionIntentFromJSONTyped,
  qt as SignedTransactionIntentToJSON,
  wo as StateApi,
  be as StateUpdatesFromJSON,
  ge as StateUpdatesFromJSONTyped,
  Oe as StateUpdatesToJSON,
  Jo as StatusApi,
  ki as SubstateBaseFromJSON,
  Ue as SubstateBaseFromJSONTyped,
  xi as SubstateBaseToJSON,
  f as SubstateFromJSON,
  ye as SubstateFromJSONTyped,
  N as SubstateIdFromJSON,
  Gt as SubstateIdFromJSONTyped,
  T as SubstateIdToJSON,
  p as SubstateToJSON,
  _r as SubstateType,
  c as SubstateTypeFromJSON,
  Ht as SubstateTypeFromJSONTyped,
  fr as SubstateTypeToJSON,
  Ei as SystemSubstateAllOfFromJSON,
  We as SystemSubstateAllOfFromJSONTyped,
  Ii as SystemSubstateAllOfToJSON,
  Dr as SystemSubstateFromJSON,
  X as SystemSubstateFromJSONTyped,
  de as SystemSubstateToJSON,
  Wn as TextApiResponse,
  Fo as TransactionApi,
  kt as TransactionHeaderFromJSON,
  xt as TransactionHeaderFromJSONTyped,
  Pt as TransactionHeaderToJSON,
  Ci as TransactionIdentifiersFromJSON,
  ze as TransactionIdentifiersFromJSONTyped,
  Di as TransactionIdentifiersToJSON,
  Et as TransactionIntentFromJSON,
  It as TransactionIntentFromJSONTyped,
  Vt as TransactionIntentToJSON,
  we as TransactionReceiptFromJSON,
  Je as TransactionReceiptFromJSONTyped,
  Fe as TransactionReceiptToJSON,
  Br as TransactionStatus,
  Ne as TransactionStatusFromJSON,
  Te as TransactionStatusFromJSONTyped,
  Hr as TransactionStatusToJSON,
  me as UpSubstateFromJSON,
  Se as UpSubstateFromJSONTyped,
  he as UpSubstateToJSON,
  Ai as V0CommittedTransactionRequestFromJSON,
  Be as V0CommittedTransactionRequestFromJSONTyped,
  He as V0CommittedTransactionRequestToJSON,
  Ge as V0CommittedTransactionResponseFromJSON,
  Le as V0CommittedTransactionResponseFromJSONTyped,
  Mi as V0CommittedTransactionResponseToJSON,
  tn as V0NetworkConfigurationResponseFromJSON,
  en as V0NetworkConfigurationResponseFromJSONTyped,
  zi as V0NetworkConfigurationResponseToJSON,
  Qe as V0NetworkConfigurationResponseVersionFromJSON,
  Xe as V0NetworkConfigurationResponseVersionFromJSONTyped,
  Ye as V0NetworkConfigurationResponseVersionToJSON,
  Ze as V0NetworkConfigurationResponseWellKnownAddressesFromJSON,
  ve as V0NetworkConfigurationResponseWellKnownAddressesFromJSONTyped,
  je as V0NetworkConfigurationResponseWellKnownAddressesToJSON,
  nn as V0StateComponentDescendentIdFromJSON,
  rn as V0StateComponentDescendentIdFromJSONTyped,
  on as V0StateComponentDescendentIdToJSON,
  Gi as V0StateComponentRequestFromJSON,
  sn as V0StateComponentRequestFromJSONTyped,
  un as V0StateComponentRequestToJSON,
  an as V0StateComponentResponseFromJSON,
  cn as V0StateComponentResponseFromJSONTyped,
  Qi as V0StateComponentResponseToJSON,
  dn as V0StateEpochResponseFromJSON,
  _n as V0StateEpochResponseFromJSONTyped,
  Yi as V0StateEpochResponseToJSON,
  vi as V0StateNonFungibleRequestFromJSON,
  fn as V0StateNonFungibleRequestFromJSONTyped,
  pn as V0StateNonFungibleRequestToJSON,
  ln as V0StateNonFungibleResponseFromJSON,
  yn as V0StateNonFungibleResponseFromJSONTyped,
  to as V0StateNonFungibleResponseToJSON,
  no as V0StatePackageRequestFromJSON,
  mn as V0StatePackageRequestFromJSONTyped,
  Sn as V0StatePackageRequestToJSON,
  hn as V0StatePackageResponseFromJSON,
  bn as V0StatePackageResponseFromJSONTyped,
  io as V0StatePackageResponseToJSON,
  so as V0StateResourceRequestFromJSON,
  gn as V0StateResourceRequestFromJSONTyped,
  On as V0StateResourceRequestToJSON,
  Nn as V0StateResourceResponseFromJSON,
  Tn as V0StateResourceResponseFromJSONTyped,
  ao as V0StateResourceResponseToJSON,
  wn as V0TransactionPayloadStatusFromJSON,
  Jn as V0TransactionPayloadStatusFromJSONTyped,
  co as V0TransactionPayloadStatusStatusEnum,
  Fn as V0TransactionPayloadStatusToJSON,
  po as V0TransactionStatusRequestFromJSON,
  Rn as V0TransactionStatusRequestFromJSONTyped,
  kn as V0TransactionStatusRequestToJSON,
  xn as V0TransactionStatusResponseFromJSON,
  Pn as V0TransactionStatusResponseFromJSONTyped,
  lo as V0TransactionStatusResponseIntentStatusEnum,
  mo as V0TransactionStatusResponseToJSON,
  ho as V0TransactionSubmitRequestFromJSON,
  En as V0TransactionSubmitRequestFromJSONTyped,
  In as V0TransactionSubmitRequestToJSON,
  Vn as V0TransactionSubmitResponseFromJSON,
  Cn as V0TransactionSubmitResponseFromJSONTyped,
  go as V0TransactionSubmitResponseToJSON,
  No as VaultSubstateAllOfFromJSON,
  Dn as VaultSubstateAllOfFromJSONTyped,
  To as VaultSubstateAllOfToJSON,
  Ur as VaultSubstateFromJSON,
  tt as VaultSubstateFromJSONTyped,
  le as VaultSubstateToJSON,
  $n as VoidApiResponse,
  Mn as canConsumeForm,
  a as exists,
  zn as instanceOfCommittedStateIdentifier,
  Lr as instanceOfCommittedTransaction,
  mr as instanceOfComponentInfoSubstate,
  Qr as instanceOfComponentInfoSubstateAllOf,
  gr as instanceOfComponentStateSubstate,
  Zr as instanceOfComponentStateSubstateAllOf,
  br as instanceOfDataStruct,
  lr as instanceOfDownSubstate,
  er as instanceOfEcdsaSecp256k1PublicKey,
  Qn as instanceOfEcdsaSecp256k1Signature,
  Yn as instanceOfEcdsaSecp256k1SignatureWithPublicKey,
  vn as instanceOfEddsaEd25519PublicKey,
  Xn as instanceOfEddsaEd25519Signature,
  jn as instanceOfEddsaEd25519SignatureWithPublicKey,
  hr as instanceOfEntityId,
  ti as instanceOfErrorResponse,
  ur as instanceOfFeeSummary,
  qr as instanceOfFungibleResourceAmount,
  ri as instanceOfFungibleResourceAmountAllOf,
  yr as instanceOfGlobalEntityId,
  Nr as instanceOfKeyValueStoreEntrySubstate,
  si as instanceOfKeyValueStoreEntrySubstateAllOf,
  wr as instanceOfNonFungibleData,
  Kr as instanceOfNonFungibleResourceAmount,
  ci as instanceOfNonFungibleResourceAmountAllOf,
  Jr as instanceOfNonFungibleSubstate,
  fi as instanceOfNonFungibleSubstateAllOf,
  sr as instanceOfNotarizedTransaction,
  Rr as instanceOfPackageSubstate,
  yi as instanceOfPackageSubstateAllOf,
  hi as instanceOfResourceAmountBase,
  Oi as instanceOfResourceChange,
  Ir as instanceOfResourceManagerSubstate,
  wi as instanceOfResourceManagerSubstateAllOf,
  xr as instanceOfResourceManagerSubstateAllOfMetadata,
  ar as instanceOfSborData,
  or as instanceOfSignedTransactionIntent,
  zr as instanceOfStateUpdates,
  Ri as instanceOfSubstateBase,
  pr as instanceOfSubstateId,
  Cr as instanceOfSystemSubstate,
  Pi as instanceOfSystemSubstateAllOf,
  rr as instanceOfTransactionHeader,
  Vi as instanceOfTransactionIdentifiers,
  ir as instanceOfTransactionIntent,
  Gr as instanceOfTransactionReceipt,
  Wr as instanceOfUpSubstate,
  qi as instanceOfV0CommittedTransactionRequest,
  Ki as instanceOfV0CommittedTransactionResponse,
  Wi as instanceOfV0NetworkConfigurationResponse,
  $i as instanceOfV0NetworkConfigurationResponseVersion,
  Ui as instanceOfV0NetworkConfigurationResponseWellKnownAddresses,
  Bi as instanceOfV0StateComponentDescendentId,
  Hi as instanceOfV0StateComponentRequest,
  Li as instanceOfV0StateComponentResponse,
  Xi as instanceOfV0StateEpochResponse,
  Zi as instanceOfV0StateNonFungibleRequest,
  ji as instanceOfV0StateNonFungibleResponse,
  eo as instanceOfV0StatePackageRequest,
  ro as instanceOfV0StatePackageResponse,
  oo as instanceOfV0StateResourceRequest,
  uo as instanceOfV0StateResourceResponse,
  _o as instanceOfV0TransactionPayloadStatus,
  fo as instanceOfV0TransactionStatusRequest,
  yo as instanceOfV0TransactionStatusResponse,
  So as instanceOfV0TransactionSubmitRequest,
  bo as instanceOfV0TransactionSubmitResponse,
  $r as instanceOfVaultSubstate,
  Oo as instanceOfVaultSubstateAllOf,
  Kn as mapValues,
  F as querystring
};
